docs/benchmarking.md

```diff
@@ -149,14 +149,98 @@ Inferences/Second vs. Client Average Batch Latency
 Concurrency: 1, throughput: 19.6095 infer/sec, latency 50951 usec
 ```
 
-## Non-supported functionalities
+### Non-supported functionalities
 
 There are a few functionalities that are missing from C API mode. They are:
 
 1. Async mode ([`--async`](cli.md#--async))
 2. For additional known non-working cases, please refer to
    [qa/L0_perf_analyzer_capi/test.sh](https://github.com/triton-inference-server/server/blob/main/qa/L0_perf_analyzer_capi/test.sh#L239-L277)
 
+
+# Benchmarking gRPC service via Dynamic gRPC
+
+> **Note**
+>
+> Dynamic gRPC service kind is under active development and currently it supports
+> bidirectional streaming RPC only at the moment.
+
+### Required steps
+
+In order to run Perf Analyzer via Dynamic gRPC service kind,
+a user needs to provide two things:
+1. A script that generates a stream of serialized Protobuf messages.
+2. An input JSON file that specifies how to execute the script.
+
+To start, write a script (for example, `input_stream.py`) that produces a sequence of Protobuf messages.
+In this example, the script serializes each message, prefixes it with a 4-byte length field,
+and writes the result to standard output.
+Ensure you install any necessary dependencies (e.g., the `protobuf` package):
+
+```python
+import sys
+# Replace with your autogenerated protobuf module, e.g.:
+# from my_protobuf_module import MyMessage
+
+def generate_msgs():
+    # Generate and yield your protobuf messages here.
+    # Example:
+    # msg = MyMessage(field1="value", field2=123)
+    # yield msg
+    yield ...   # Replace with actual message generation logic
+    yield ...   # Add more messages as required
+
+for msg in generate_msgs():
+    serialized = msg.SerializeToString()
+    # Write the message length as 4-byte integer (using system byte order)
+    sys.stdout.buffer.write(len(serialized).to_bytes(4, byteorder=sys.byteorder))
+    # Write the serialized message itself
+    sys.stdout.buffer.write(serialized)
+    sys.stdout.buffer.flush()
+```
+
+Next, create an input JSON file (for example, `inputs.json`) that instructs Perf Analyzer to run your script.
+Perf Analyzer will execute the command specified and use the output as the stream of Protobuf messages:
+
+```json
+{
+  "data": [
+    {
+      "ipc_stream": "python3 input_stream.py"
+    }
+  ]
+}
+```
+(Note: The "ipc_stream" key holds the command that will be executed to generate the messages.)
+
+With both the generator script and the JSON configuration in place,
+run Perf Analyzer using the Dynamic gRPC service kind.
+Replace `<URL>`, `<package>`, `<service>`, and `<method>` with the appropriate values for your gRPC service:
+
+```bash
+perf_analyzer --service-kind=dynamic_grpc -u=<URL> --input-data=inputs.json  --rpc=<package>.<service>/<method> --streaming
```

Comment:
Also, regarding the `--rpc` option name, it's referring to the fully qualified method name, right?

Should the option name reflect that more clearly? Is "rpc" a community-common shorthand for a fully qualified method name? Maybe `--grpc-method`?